grpclog/component.go

```diff
@@ -0,0 +1,162 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpclog
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"google.golang.org/grpc/internal/grpclog"
+)
+
+// componentData records the settings for a component.
+type componentData struct {
+	name      string
+	verbosity int
+	level     int
+}
+
+const sentinel = math.MinInt32
+
+// log levels
+const (
+	levelInfo = iota
+	levelWarning
+	levelError
+)
+
+var environmentVars = map[string]*componentData{}
+var prefixVars = map[string]*componentData{}
+var cache = map[string]*componentData{}
+
+// apply applies the parameter componentData to the receiver componentData.
+// Sentinel values in the parameter will not be applied.
+func (c *componentData) apply(applyData *componentData) {
+	if applyData.verbosity != sentinel {
+		c.verbosity = applyData.verbosity
+	}
+	if applyData.level != sentinel {
+		c.level = applyData.level
+	}
+}
+
+func (c *componentData) InfoDepth(depth int, args ...interface{}) {
+	if c.level > levelInfo {
```

Comment:
No need to check this. Component is just a naive wrapper to prepend the component name. It shouldn't try to control logs.

Reply:
Removed level/verbosity and all log control logic from component logging.